### PR TITLE
[ADD] website : generic template for search and add functionality to …

### PR DESCRIPTION
**Pull request by ajayjaviya**
Task:https://www.odoo.com/web#id=13284&view_type=form&model=project.task&menu_id=3942&action=327
Pad: https://pad.odoo.com/p/r.Cy7i8H4sFcoNZ89Y

### CODE REVIEW

File: addons/website/static/src/js/website.js [hunk 1]
```diff
@@ -278,6 +278,15 @@ $(document).ready(function () {
...
     if($.fn.placeholder) $('input, textarea').placeholder();
+      $(".oe_search_box").on('input', function() {
+          $(this).next('span').toggle(!!$(this).val());
+      });
+      $(".oe_search_clear").toggle(!!$(this).prev('input').val())
+          .on('click', function() {
+              $(this).prev('input').val('').focus();
+              $(".oe_search_button").trigger("click");
+          });
+      $(".oe_search_clear").toggle(!!$('.oe_search_box').val());
```
Comment (Gorash) [inline addons/website/static/src/js/website.js]: Why toggle twice ? the first is useless

Comment (aja-odoo) [inline addons/website/static/src/js/website.js]: remove it

**Comment by Gorash** [comment]
@tde-banana-odoo you can merge

File: addons/website/static/src/js/website.js [hunk 2]
```diff
@@ -278,6 +278,14 @@ $(document).ready(function () {
     dom_ready.resolve();
     // fix for ie
     if($.fn.placeholder) $('input, textarea').placeholder();
+      $(".oe_search_box").on('input', function() {
```
Comment (tde-banana-odoo) [inline addons/website/static/src/js/website.js]: Why is the indentation incorrect ? Does this code belong to the previous if ?

File: addons/website/views/website_templates.xml [hunk 3]
```diff
@@ -1102,5 +1102,16 @@ Sitemap: <t t-esc="url_root"/>sitemap.xml
     </t>
 </template>
 
+<template id="search" name="search">
```
Comment (tde-banana-odoo) [inline addons/website/views/website_templates.xml]: Do not use such standard name as "search". Please. If someday you have to grep the code, what will happen do you think ?

File: addons/website_membership/controllers/main.py [hunk 4]
```diff
@@ -33,7 +33,7 @@ def members(self, membership_id=None, country_name=None, country_id=0, page=1, *
         country_obj = request.registry['res.country']
         membership_line_obj = request.registry['membership.membership_line']
         partner_obj = request.registry['res.partner']
-        post_name = post.get('name', '')
+        post_name = post.get('search', '')
```
Comment (tde-banana-odoo) [inline addons/website_membership/controllers/main.py]: Maybe add backward compatibility by checking name if search does not exist.

File: addons/website/static/src/js/website.js (hunk 2 above)
Comment (aja-odoo) [inline addons/website/static/src/js/website.js]: i have correct a indentation

File: addons/website/views/website_templates.xml (hunk 3 above)
Comment (aja-odoo) [inline addons/website/views/website_templates.xml]: done, change it to search_box is it ok ?

File: addons/website_membership/controllers/main.py (hunk 4 above)
Comment (aja-odoo) [inline addons/website_membership/controllers/main.py]: done

File: addons/website/views/website_templates.xml [hunk 5]
```diff
@@ -1102,5 +1102,16 @@ Sitemap: <t t-esc="url_root"/>sitemap.xml
     </t>
 </template>
 
+<template id="search_box" name="Searchbox">
```
Comment (tde-banana-odoo) [inline addons/website/views/website_templates.xml]: Just rename to website_search_box (module prefix to ease the grep).

Comment (aja-odoo) [inline addons/website/views/website_templates.xml]: done

**Comment by Gorash** [comment]
merged

**Pull request closed without merging**